lib/chef/resource/ohai_hint.rb

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright:: Copyright 2011-2018, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class OhaiHint < Chef::Resource
+      resource_name :ohai_hint
+      provides :ohai_hint
+
+      description "Use the ohai_hint resource to pass hint data to Ohai to aid in configuration detection."
```

Comment:
Don't think we should repeat the name of the resource in the description. We could probably use some wordsmithing and advice from @brewn here. (And on the rest of these).

Reply:
This goes off the format on the docs site. Here's an example:

Use the apt_repository resource to specify additional APT repositories. Adding a new repository will update APT package cache immediately.